config/macos_clang_arm64.mk

```diff
@@ -0,0 +1,23 @@
+BUILDDIR:=macos/clang/x86_64
+
+include config/base.mk
+include config/with-clang.mk
+include config/with-debug.mk
+include config/with-brutality.mk
+include config/with-optimization.mk
+include config/with-threads.mk
+
+# Clang sadly doesn't support important optimizations.  This practically
+# limits clang usage to code hygenine usage for the time being.  Here,
+# ideally would do:
+#
+# -falign-functions=32 -falign-jumps=32 -falign-labels=32 -falign-loops=32
+# -mbranch-cost=5
+
+CPPFLAGS+=-fomit-frame-pointer -march=native -mtune=skylake \
+	  -DFD_HAS_INT128=1 -DFD_HAS_DOUBLE=1 -DFD_HAS_ALLOCA=1 -DFD_HAS_X86=1
+
+FD_HAS_INT128:=1
+FD_HAS_DOUBLE:=1
+FD_HAS_ALLOCA:=1
+FD_HAS_X86:=1
```

Comment:
As above, the reason for using x64/x86 flags in the macOS ARM Makefile is because I was testing this on my macOS M2 machine using Rosetta. Also as above, I can work on making this build & run correctly without Rosetta, though this will seemingly become https://github.com/firedancer-io/firedancer/issues/99